eslint.config.mjs

```diff
@@ -0,0 +1,127 @@
+import eslint from '@eslint/js'
+import vitest from '@vitest/eslint-plugin'
+import importPlugin from 'eslint-plugin-import'
+import jestDom from 'eslint-plugin-jest-dom'
+import react from 'eslint-plugin-react'
+import reactCompiler from 'eslint-plugin-react-compiler'
+import reactHooks from 'eslint-plugin-react-hooks'
+import testingLibrary from 'eslint-plugin-testing-library'
+import globals from 'globals'
+import tseslint from 'typescript-eslint'
+
+export default tseslint.config(
+  {
+    ignores: ['**/dist/'],
+  },
+  eslint.configs.recommended,
+  tseslint.configs.recommended,
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        ...globals.es2015,
+      },
```

Comment:
I'm still concerning this, but I might be misunderstand something, so I'll take it over.

Reply:
<img width="649" alt="image" src="https://github.com/user-attachments/assets/ee345366-87bd-4503-a2d4-109dcbd0a555" />

this is jotal-valtio code, as you see, eslint don't know where `console` comes from. so we have to sepecify with `globals`

Reply:
thanks for confirming. there might be some misconfiguration in my editor. i'll look into it!

Reply:
* https://github.com/sindresorhus/globals/blob/main/data/browser.mjs
* https://github.com/sindresorhus/globals/blob/main/data/node.mjs
  * https://github.com/sindresorhus/globals/blob/main/data/nodeBuiltin.mjs
  * https://github.com/sindresorhus/globals/blob/main/data/commonjs.mjs
* https://github.com/sindresorhus/globals/blob/main/data/es2015.mjs

@dai-shi These code snippets will help you understand.

Reply:
I don't think I'm misunderstanding what it is. I'm just against to it.

Reply:
> As you can see, this is the https://github.com/jotaijs/jotai-valtio/pull/11. ESLint does not know the source of console.

Yeah, I can reproduce it in eslint.config.js. But, if I write console.log in src/index.ts, it doesn't complain. I expect it to complain without `globals`. Can you help dig into it? It feels like `extensions` works for something and doesn't work for something else.

Reply:
@dai-shi 

I think `js` and `jsx` rely on `globals` setting for recognizing global variables, whereas `ts` and `tsx` already have global type support provided by TypeScript, making the setting unnecessary. I also tested it, and the error only occurred in `js`.

So, I think in the top-level `eslint.config.mjs`, our main logic in the `ts` and `tsx` files doesn’t need `globals`, and since the purpose of the `js` file is for configuration, setting only `globals.node` should be enough.

Reply:
> <img alt="image" width="649" src="https://private-user-images.githubusercontent.com/66409882/398609599-ee345366-87bd-4503-a2d4-109dcbd0a555.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUyMTIwODksIm5iZiI6MTczNTIxMTc4OSwicGF0aCI6Ii82NjQwOTg4Mi8zOTg2MDk1OTktZWUzNDUzNjYtODdiZC00NTAzLWEyZDQtMTA5ZGNiZDBhNTU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI2VDExMTYyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjY2M1ZTZiNzFmM2I5ZDZhNmJkZTNjOGIxYTA3NjdmMGVmZDA0OGJlYWIzMjljZTRiZGJhNWYwNWNmZTU2NTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0CipveMgvy5F79veZeQ1pEEZ_cpBkrw1WKMYykTSYeU">
> @dai-shi As you can see, this is the [`jotal-valtio`](https://github.com/jotaijs/jotai-valtio/pull/11). ESLint does not know the source of `console`. Therefore, we need to inform ESLint of its source through `globals`.

@dai-shi Also this was a`js` file

Reply:
Hmm, alright, that's not what I originally proposed, but if it's already configured, I wouldn't rather disable it.

We basically use ts/tsx only. So,

> since the purpose of the `js` file is for configuration, setting only `globals.node` should be enough.

Yeah, and we don't use `console` in config files, so `globals.node` isn't necessary. (If we really need it, we can reconsider or use inline config.)

Reply:
@dai-shi I gonna remove `globals`